geoutils/georaster.py

```diff
@@ -1440,20 +1444,43 @@ def interp_points(self,pts,nsize=1,mode='linear',band=1):
 
         return rpts
 
-    def split_bands(self):
+    def split_bands(self, copy=False, subset: Optional[Union[list[int], int]] = None):
         """
         Split the bands into separate copied rasters.
 
-        :returns: A list of Rasters for each band.
+        :param copy: Copy the bands or return slices of the original data.
+        :param subset: Optional. A subset of band indices to extract. Defaults to all.
+
+        :returns: A list of Rasters for each band, or one Raster if len(subset)==1.
         """
         bands: list[self] = []
 
-        for band_n in range(self.nbands):
-            bands.append(self.from_array(
-                self.data[band_n, :, :],
-                transform=self.transform,
-                crs=self.crs,
-                nodata=self.nodata
-            ))
-
-        return bands
+        if subset is None:
+            indices = list(range(self.nbands))
+        elif isinstance(subset, int):
+            indices = [subset]
+        elif isinstance(subset, list):
+            indices = subset
+        else:
+            raise ValueError(f"'subset' got invalid type: {type(subset)}. Expected list[int], int or None")
+
+        if copy:
+            for band_n in indices:
+                # Generate a new Raster from a copy of the band's data
+                bands.append(self.from_array(
+                    self.data[band_n, :, :],
+                    transform=self.transform,
+                    crs=self.crs,
+                    nodata=self.nodata
+                ))
+        else:
+            for band_n in indices:
+                # Generate a new instance with the same underlying values.
+                raster = Raster(self)
+                # Set the data to a slice of the original array
+                raster._data = self.data[band_n, :, :].reshape(((1,) + self.data.shape[1:]))
```

Comment:
What if indices is only a subset of the original bands? 
Let's say e.g. self had 3 bands originally, and you extract only 1 band. Then you will loop only once, but self.data still hasa length of 3, no?

Reply:
Oh no, I think I get it. The output is always a single band raster.
But does this effectively not copy the data??

Reply:
There's no copying! Check the tests! It's black magic.